tsdb/index/index.go

```diff
@@ -1683,6 +1708,36 @@ func (dec *Decoder) Postings(b []byte) (int, Postings, error) {
 	return n, newBigEndianPostings(l), d.Err()
 }
 
+// LabelValueFor decodes a label for a given series.
+func (dec *Decoder) LabelValueFor(b []byte, label string) (string, error) {
+	d := encoding.Decbuf{B: b}
+	k := d.Uvarint()
+
+	for i := 0; i < k; i++ {
+		lno := uint32(d.Uvarint())
+		lvo := uint32(d.Uvarint())
+
+		if d.Err() != nil {
+			return "", errors.Wrap(d.Err(), "read series label offsets")
+		}
+
+		ln, err := dec.LookupSymbol(lno)
+		if err != nil {
+			return "", errors.Wrap(err, "lookup label name")
+		}
+		lv, err := dec.LookupSymbol(lvo)
+		if err != nil {
+			return "", errors.Wrap(err, "lookup label value")
+		}
```

Comment:
This should go inside `if ln == label {`

Reply:
thx, fixed cdbc0027f957a0c4329db915c998f25371d920dd